api/server/httputils/errors.go

```diff
@@ -54,7 +54,7 @@ func GetHTTPErrorStatusCode(err error) int {
 			code    int
 		}{
 			{"not found", http.StatusNotFound},
-			{"no such", http.StatusNotFound},
+			{"no such", http.StatusBadRequest},
```

Comment:
This is not correct, because an image/container/network/xyz "not found" is okay to return a 404.

The discussion was if `Error parsing reference: "qweqi,weqw" is not a valid repository....` should return a different code (so only if the specified reference was using an invalid format